src/jsdoc_transformer.ts

```diff
@@ -526,21 +541,38 @@ export function jsdocTransformer(
       }
 
       /** Function declarations are emitted as they are, with only JSDoc added. */
-      function addJsDocToFunctionLikeDeclaration(fnDecl: ts.FunctionLikeDeclaration) {
+      function visitFunctionLikeDeclaration(fnDecl: ts.FunctionLikeDeclaration) {
         if (!fnDecl.body) {
           // Two cases: abstract methods and overloaded methods/functions.
           // Abstract methods are handled in emitTypeAnnotationsHandler.
           // Overloads are union-ized into the shared type in FunctionType.
-          return;
+          return ts.visitEachChild(fnDecl, visitor, context);
         }
         const extraTags = [];
         if (hasExportingDecorator(fnDecl, typeChecker)) extraTags.push({tagName: 'export'});
 
-        const [tags, ] = moduleTypeTranslator.getFunctionTypeJSDoc([fnDecl], extraTags);
+        const {tags, thisReturnType} =
+            moduleTypeTranslator.getFunctionTypeJSDoc([fnDecl], extraTags);
         const mjsdoc = moduleTypeTranslator.getMutableJSDoc(fnDecl);
         mjsdoc.tags = tags;
         mjsdoc.updateComment();
         moduleTypeTranslator.blacklistTypeParameters(fnDecl, fnDecl.typeParameters);
+
+        const contextThisTypeBackup = contextThisType;
+        contextThisType = thisReturnType;
+        const result = ts.visitEachChild(fnDecl, visitor, context);
+        contextThisType = contextThisTypeBackup;
+        return result;
+      }
+
+      /**
+       * In methods with a templated this type, adds explicit casts to accesses on this.
+       *
+       * @see contextThisType
+       */
+      function visitThisExpression(node: ts.ThisExpression) {
+        if (!contextThisType) return ts.visitEachChild(node, visitor, context);
```

Comment:
What are the children of a this expression?

Reply:
None. But I think it's better to follow the general pattern instead of special casing this per syntax construct, isn't it?